app/js/Router.js

```diff
@@ -19,76 +8,94 @@ const SmokeTest = require('smoke-test-sharelatex')
 const AuthController = require('./AuthController')
 const HttpController = require('./HttpController')
 const SessionMiddleware = require('./SessionMiddleware')
+const { rateLimit } = require('./RateLimitMiddleware')
 
-module.exports = Router = {
-  initialize(app) {
-    app.use(express.static('public'))
-    app.use(SessionMiddleware.middleware)
-    app.use(BodyParser.urlencoded({ extended: false }))
+module.exports = {
+  initialize
+}
+
+function initialize(app) {
+  app.use(express.static('public'))
+  app.use(SessionMiddleware.middleware)
+  app.use(BodyParser.urlencoded({ extended: false }))
 
-    app.get('/', HttpController.home)
+  app.get('/', HttpController.home)
 
-    app.post(
-      '/login',
-      celebrate({
-        body: Joi.object({
-          email: Joi.string()
-            .trim()
-            .email()
-            .required(),
-          password: Joi.string()
-            .trim()
-            .required()
-        })
-      }),
-      AuthController.login,
-      AuthController.handleLoginErrors
-    )
+  app.post(
+    '/login',
+    celebrate({
```

Comment:
Any reason not to put the rate limiting as the first middleware, before the validation? Appreciate it doesn't really make a difference here but seems like good practice.

Reply:
I was thinking that validation errors don't count as login attempts, but as you say, it doesn't make much of a difference. I can move it.

Reply:
Actually, now that I think about it, since we're trying to mitigate problems with institutional proxies, it might be better to keep it that way? It potentially reduces the number of requests that count as login attempts and doesn't add to the security risk as far as I can tell.

Reply:
I think it's fine to leave it. 

I think we once got burned by having some middleware before the rate limiter in web (might have been the bodyparser or something like that) and the middleware was doing something that turned out to be expensive, which we only found out when something started hitting it hard.  The solution was moving the rate-limiting to the front.  I always get a  feeling of something bad happening when I see rate-limiting not at the front now anyway.